### PR TITLE
tab: fix active bar bug (#14416)

### LINKED ISSUE #14351: [Bug Report] Tabs custom label cannot be modified immediately after changing label
### Element UI version
2.5.4

### OS/Browsers version
window 10

### Vue version
2.5.17

### Reproduction Link
[https://jsfiddle.net/xiaoniezi/gvn0ur3x/](https://jsfiddle.net/xiaoniezi/gvn0ur3x/)

### Steps to reproduce
1.tab自定义label

2.延迟修改label的内容

3.点击tab

### What is Expected?
可以立即触发label相应的内容

### What is actually happening?
需要点击tab才会触发

### LINKED ISSUE #14416: [Bug Report] Tab active bar doesn't resize when tab label changes dynamically
### Element UI version
2.5.4

### OS/Browsers version
Windows 7 / Chrome 72.0.3626.109

### Vue version
2.6.6

### Reproduction Link
[https://jsfiddle.net/akki_jat/quce1ntd/2/](https://jsfiddle.net/akki_jat/quce1ntd/2/)

### Steps to reproduce
Just run jsfiddle and observe

### What is Expected?
Resize the tab active bar as label changes dynamically

### What is actually happening?
Tab active bar is not resizing as label changes dynamically

### PR DESCRIPTION
**Pull request by akki-jat**
fix: Tab active bar doesn't resize when tab label changes dynamically

Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

- Closes #14416 
- Closes #14351

### CODE REVIEW

**Comment by akki-jat** [comment]
@Leopoldthecoder, @ziyoung, @wacky6, @jikkai can anyone look at this?

File: packages/tabs/src/tabs.vue [hunk 1]
```diff
@@ -172,14 +172,24 @@
...
       }
+
+      this.$on('tabLabelChanged', () => this.calcPaneInstances(true));
     },
 
     mounted() {
       this.calcPaneInstances();
     },
 
+    /**
```
Comment (ziyoung) [inline packages/tabs/src/tabs.vue]: It's needed. See pr #12846
Remove comment.

File: packages/tabs/src/tabs.vue [hunk 2]
```diff
@@ -172,14 +172,24 @@
       if (!this.currentName) {
         this.setCurrentName('0');
       }
+
+      this.$on('tabLabelChanged', () => this.calcPaneInstances(true));
```
Comment (ziyoung) [inline packages/tabs/src/tabs.vue]: You can use bind  and it's concise. 
```js
this.$on('tabLabelChanged', this.calcPaneInstances.bind(true));
```

File: packages/tabs/src/tabs.vue [hunk 3]
```diff
@@ -55,13 +55,13 @@
...
     methods: {
-      calcPaneInstances() {
+      calcPaneInstances(isLabelUpdated = false) {
         if (this.$slots.default) {
           const paneSlots = this.$slots.default.filter(vnode => vnode.tag &&
             vnode.componentOptions && vnode.componentOptions.Ctor.options.name === 'ElTabPane');
           // update indeed
           const panes = paneSlots.map(({ componentInstance }) => componentInstance);
-          if (!(panes.length === this.panes.length && panes.every((pane, index) => pane === this.panes[index]))) {
+          if (isLabelUpdated || !(panes.length === this.panes.length && panes.every((pane, index) => pane === this.panes[index]))) {
```
Comment (ziyoung) [inline packages/tabs/src/tabs.vue]: This line is too long. We can declare the second expression as a variable.
```js
const panesChanged = !(panes.length === this.panes.length && panes.every((pane, index) => pane === this.panes[index]))
if (isLabelUpdated || panesChanged) {
// ...
}
```

File: packages/tabs/src/tabs.vue [hunk 4]
```diff
@@ -172,14 +172,24 @@
...
 
+    /**
+     * ? Is this method still needed - if this method's work is only update when label is chnaged,
+     * then should be removed becaused it's handled by event
+     */
     updated() {
       this.calcPaneInstances();
+    },
+
+    beforeDestroy() {
```
Comment (ziyoung) [inline packages/tabs/src/tabs.vue]: This hook is redundant.

**Comment by ziyoung** [comment]
@akki-jat update your pr. I will merge it soon.

**Comment by akki-jat** [comment]
> @akki-jat update your pr. I will merge it soon.

@ziyoung  take a look now.

**Pull request merged**